### PR TITLE
Added tags to lead list filters

### LINKED ISSUE #840: Smart lists - add filter by tags
New features in 1.2 beta are useful, but it will be great also add filter by tags.

### PR DESCRIPTION
**Pull request by alanhartless**
**Description**

This PR adds tags as a filter to lead lists.

**Testing**
Apply the PR.  Run `doctrine:migrations:migrate` command (just adds a new index).  Then create a new list and choose Tags as a filter.  Select or create the tags to use. Tag a lead with the same tag and run the `mautic:list:update` command.  The leads should be added to the list.

Resolves #840 

Note: I merged the release-1.2.0 branch into this branch to get the latest migrations so it's showing up in the history.

### CODE REVIEW

**Comment by escopecz** [comment]
tl;dr: The problem is not related with this PR.

Options doesn't show up for me. Even though I can see the select html tag in the source code with options in and not in. Console doesn't display any JS error. Clearing browser cache didn't help.
tag-select
This problem isn't only in the tag operator dropdown, but others as well. Tested on Ubuntu, Chrome. FireFox works.

[... some time passed ...]

I tried to remove attributes from the select tag, disable some or all CSS styles. Nothing helped. Then I tried to use the ******* Google and found http://askubuntu.com/questions/641098/blank-select-options-on-chrome-chromium-ubuntu-14-04.
This issue happens only if the browser is in external monitor. What a dumb bug. So I moved Chrome to my main monitor and tested the rest of the workflow :)

The test went well and I vote up :+1:

**Comment by dbhurley** [comment]
Very minor thing, but what do others think of replacing "in" and "not in" with "including" and "excluding" that way the line reads: tags including .... and tags excluding ....

**Comment by escopecz** [comment]
That's better wording I think. I wonder what would Brian suggest...

**Comment by dbhurley** [comment]
This worked perfect. +1
